practice/completer/site.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+/**
+ *  The Completer Constructor holds valid words that can be called through
+ * the prototype method complete, to find valid words that start with given
+ * prefix.
+ * There is also a weighting system, where everytime selectCompletion is called
+ * the selected string has weight added to it. The complete function returns
+ * an array ordered in more weighted words first.
+ */
+function Completer() {
+  this.validCompletionToWeight = {};
+}
+
+var completerProto = {
+  addCompletion: function(str) {
```

Comment:
Unfortunately the linter doesn't catch this, but all functions given names need JSDocs, even if they're defined in this other way.